File: Algorithms/868.Binary-Gap/binary_gap.py
class Solution:
    def binaryGap(self, N: int) -> int:
        res = 0
        dis = 0
        flag = False
        while N:
            if N % 2 == 1:
                if flag:
                    res = max(res, dis)
                else:
                    flag = True
                dis = 0
            dis += 1
            N = N // 2
        return res

File: Algorithms/868.Binary-Gap/binary_gap.go
func binaryGap(N int) int {
    for N != 0 && N & 1 != 1 {
        N = N >> 1
    }
    
    max_dis := 0
    consecutive := 0
    for N > 0 {
        N = N / 2
        consecutive += 1
        if N & 1 == 1 {
            if max_dis < consecutive {
                max_dis = consecutive
            }
            consecutive = 0
        }
    }
    return max_dis
}